### ISSUE TITLE
Omitted exception on deletion of connected node in transaction

### ISSUE THREAD

**Issue by fickludd** [neo4j/neo4j#13341]
When executing multiple cypher queries in the same transaction, deletion of connected node does not throw an error until commit().

**Neo4j Version:** 4.4.11 (community)
**Operating System:** Ubuntu
**API:** java driver

### Steps to reproduce

Execute the following java/scala sequence in a test
```
test("apa") {
    val tx = sessionProvider.session().beginTransaction()
    tx.run("CREATE (a:A)-[:R]->(b:B)").consume()
    tx.run("MATCH (a:A) DELETE a").consume()
    tx.rollback()
}
```

### Expected behavior
An exception throw on the `DELETE` query, as `a` has at least one relationship. 

### Actual behavior
No exception thrown.
Alternatively, if we try to `commit()` the transaction a mysterious exception 
  `org.neo4j.driver.exceptions.DatabaseException: Unable to complete transaction.`
is thrown on commit.

**Comment by emmaholmbergohlsson** [comment]
@fickludd Sorry for the long response time, we lost this among other cards for a while
This is by design. It is allowed to remove those relationships later on in the same transaction which would make it possible to commit. As far as I know there are no official guarantees of when the delete should throw, but the current behaviour is at commit, and it is not likely to change, especially not on 4.4.

**Comment by fickludd** [comment]
@emmaholmbergohlsson Thanks for the reply :).

I get that it might not change, but I think it's makes for a quite poor user experience and an also makes it impossible to rely on the delete check for user feedback. At minimum it would be helpful to get the same error message on commit, as opposed to getting no explanation.

**Comment by emmaholmbergohlsson** [comment]
@fickludd Yeah I can definitely see this being annoying from a user standpoint. I have questioned it working that way more than once.
Like I said I don't expect it to change any time soon, but I'll bring it higher up for a discussion.

**Comment by emmaholmbergohlsson** [comment]
@fickludd I've now checked what the exception is on dev, apparently I improved it myself last summer when I ran into it 😅
You will get either `org.neo4j.graphdb.ConstraintViolationException: Cannot delete node that was created in this transaction, because it still has relationships.`, or `org.neo4j.graphdb.ConstraintViolationException: Cannot delete node<0>, because it still has relationships. To delete this node, you must first delete its relationships.` if the node was created in an earlier transaction.

Try out a newer version of Neo4j if you'd like to get this improved exception instead

**Comment by fickludd** [comment]
Nice, thanks for checking and fixing 👍
